File: 0021-merge-two-sorted-lists/0021-merge-two-sorted-lists.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode mergeTwoLists(ListNode list1, ListNode list2) {
        if (list1 == null){
            return list2;
        }
        if (list2 == null ){
            return list1;
        }
        ListNode tHead = new ListNode(-1);
        ListNode curr = tHead;
        ListNode c1 = list1;
        ListNode c2 = list2;
        while (c1 != null && c2 != null){
            if (c1.val < c2.val){
                curr.next = c1;
                c1 = c1.next;
            } else{
                curr.next = c2;
                c2 = c2.next;
            }
            curr = curr.next;
        }
        while (c1 != null){
            curr.next = c1;
            c1 = c1.next;
            curr = curr.next;
        }

        while (c2 != null){
            curr.next = c2;
            c2 = c2.next;
            curr = curr.next;
        }
        return tHead.next;
    }
}

File: 0021-merge-two-sorted-lists/0021-merge-two-sorted-lists.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* mergeTwoLists(ListNode* list1, ListNode* list2) {
        ListNode *tHead = new ListNode(-1);
        ListNode *curr = tHead;
        while (list1 && list2){
            if (list1->val > list2->val){
                curr->next = list2;
                list2 = list2->next;
            } else {
                curr->next = list1;
                list1 = list1->next;
            }   
            curr = curr->next;    
        }
        while (list1){
            curr->next = list1;
            list1 = list1->next;
            curr = curr->next;
        }
        while (list2){
            curr->next = list2;
            list2 = list2->next;
            curr = curr->next;
        }
        return tHead->next;
    }
};